lib/pdk/util.rb

```diff
@@ -8,6 +8,14 @@
 
 module PDK
   module Util
+    PACKAGED_TEMPLATE_KEYWORD = 'pdk-default'.freeze
```

Comment:
I'd use `'https://github.com/puppetlabs/pdk-module-template'` as `PACKAGED_TEMPLATE_KEYWORD`. This allows users to understand what's happening and where to go, while to the PDK it's just another string.

Reply:
I think using `https://github.com/puppetlabs/pdk-module-template` (an old, deprecated repo for templates) would be confusing for users. I also didn't want a string that was clearly a URI or filepath, because we need to disambiguate the custom templates, gem installs, and package installs.

Reply:
:man_facepalming: yeah that was a copy/paste error. I meant to put in the current templates URL

Reply:
now I'm curios: what difference does it make to the module if the PDK binary is from a gem or a package?

Reply:
More related to the work done in #434 , but one example is when performing an `update` on a package install that uses the packaged templates, we will need to overlook the `tag` in template-ref and proceed with an update to the latest tag. On a gem install, 1) the package templates path will not exist and 2) any `tag` ref will need to be honored as a static pin and only changes to .sync.yml will be processed until the user changes the `template-ref` to something else.

Reply:
I see now. Does that come from overloading `template-ref` in the `metadata.json` with "the ref used for the last update" and "the ref that should be used for updating"? Splitting that up and making it explicit (e.g. using `last-update-ref` and `next-update-ref`) might make those issues go away. #434 seems to be a good place to make that change.

Reply:
After chatting offline about this, I now get your point: When running from the native packages, and using the built-in pdk-default templates, this needs to specifically NOT be handled like a cache, but a separate source, because the pdk-default templates are pinned to a very specific version.

Reply:
In retrospect, I'm wondering if the `PACKAGED_TEMPLATE_KEYWORD` should be `pdk-packaged` instead. To avoid confusion about defaults.

/cc @DavidS @rodjek @scotje

Reply:
"packaged-templates" ? #bikeshed